src/prefect/core/flow.py

```diff
@@ -994,6 +994,13 @@ def _run(
             "context", {}
         ).copy()  # copy to avoid modification
 
+        # set flow_run_id from args or uuid if flow_run_id is not an argument
+        flow_run_id = kwargs.pop("flow_run_id", None)
+        if flow_run_id is not None:
+            flow_run_context.setdefault("flow_run_id", flow_run_id)
+        else:
+            flow_run_context.setdefault("flow_run_id", str(uuid.uuid4()))
```

Comment:
This one line is actually sufficient, without popping or if / else statements

Reply:
great! I changed the 1 if to a one line, still need to use the pop to get the argument out of kwargs without changing the function signature